org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/resources/SonarLintPropertyTester.java

```diff
@@ -51,9 +51,11 @@ private static IProject getProject(Object receiver) {
       return ((IResource) receiver).getProject();
     }
     if (receiver instanceof IAdaptable) {
-      IProject project = ((IAdaptable) receiver).getAdapter(IProject.class);
+      // note: the cast to IProject is necessary for e43 and e44
+      IProject project = (IProject) ((IAdaptable) receiver).getAdapter(IProject.class);
       if (project == null) {
-        IResource res = ((IAdaptable) receiver).getAdapter(IResource.class);
+        // note: the cast to IResource is necessary for e43 and e44
+        IResource res = (IResource) ((IAdaptable) receiver).getAdapter(IResource.class);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unnecessary cast to "IResource". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1905)